parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -38,44 +48,87 @@ public void updateStats(Binary value) {
   public void mergeStatisticsMinMax(Statistics stats) {
     BinaryStatistics binaryStats = (BinaryStatistics)stats;
     if (!this.hasNonNullValue()) {
-      initializeStats(binaryStats.getMin(), binaryStats.getMax());
+      initializeStats(binaryStats.genericGetMinSigned(), binaryStats.genericGetMaxSigned());
+      initializeStats(binaryStats.genericGetMinUnsigned(), binaryStats.genericGetMaxUnsigned());
```

Comment:
good point, this should separately try and merge all 3 pairs of fields when they are available